### PR TITLE
feat: Add github actions for Revive and GolangCI-Lint

### PR DESCRIPTION
**Pull request by matoous**
Add Github Actions and Makefile Targets for _GolangCI lint_ and _Revive_ lint, refactor the `lint` makefile target to do both. This should make it easier (after @fallion figures out how to handle automatic updates to Github actions) to maintain this as a Github repo without many external dependencies.

Signed-off-by: Matous Dzivjak <matousdzivjak@gmail.com>

### CODE REVIEW

File: .golangci.yml [hunk 1]
```diff
@@ -34,13 +34,17 @@ linters-settings:
...
+    lines: 70
+    statements: 60
 
 linters:
   enable-all: true
   disable:
     # prealloc is not recommended by `golangci-lint` developers.
     - prealloc
     - gochecknoglobals
+    - wsl
```
Comment (nglgzz) [inline .golangci.yml]: Why is this linter disabled?

File: api/rest/middleware_deprecation_test.go [hunk 2]
```diff
@@ -9,6 +9,7 @@ import (
 	"github.com/stretchr/testify/assert"
 )
 
+//revive:disable // for testing purposes
```
Comment (nglgzz) [inline api/rest/middleware_deprecation_test.go]: Can you please add which rule was disabled here?

File: .golangci.yml (hunk 1 above)
Comment (matoous) [reply .golangci.yml]: Fixed, added a comment why. You can check it yourself as well, `wsl` force blank lines on many places where they are not necessary (that's an opinion so feel free to take a look yourself).

File: api/rest/middleware_deprecation_test.go (hunk 2 above)
Comment (matoous) [reply api/rest/middleware_deprecation_test.go]: Added a comment.

**Pull request merged**